### PR TITLE
feat(HTTP Request Node): Follow redirects by default

### PR DESCRIPTION
**Pull request by agobrech**

### CODE REVIEW

File: packages/nodes-base/nodes/HttpRequest/V3/HttpRequestV3.node.ts [hunk 1]
```diff
@@ -1038,14 +1038,13 @@ export class HttpRequestV3 implements INodeType {
 			if (autoDetectResponseFormat || fullResponse) {
 				requestOptions.resolveWithFullResponse = true;
 			}
-
-			if (redirect?.redirect?.followRedirects) {
+			if (redirect?.redirect?.followRedirects || (nodeVersion >= 5 && redirect === undefined)) {
```
Comment (netroy) [inline packages/nodes-base/nodes/HttpRequest/V3/HttpRequestV3.node.ts:1041]: You could extract the repeated check into a separate variable
```typescript 
const autoRedirect = nodeVersion >= 5 && redirect === undefined;
if (autoRedirect || redirect?.redirect?.followRedirects) {
```

**Review by netroy — DISMISSED** [review]
besides the one comment, LGTM

**Review by csuermann — COMMENTED** [review]
How would this PR affect the user experience in the UI? The HTTP Request Node already has an option toggle for following redirects. I would somehow expect this toggle to be set by default in v5?!

**Comment by agobrech** [comment]
> How would this PR affect the user experience in the UI? The HTTP Request Node already has an option toggle for following redirects. I would somehow expect this toggle to be set by default in v5?!

With this PR if the toggle is not in the UI it will follow redirects, however if you add the toggle it will already be set as false.

File: packages/nodes-base/nodes/HttpRequest/HttpRequest.node.ts [hunk 2]
```diff
@@ -14,14 +14,15 @@ export class HttpRequest extends VersionedNodeType {
...
-			defaultVersion: 4,
+			defaultVersion: 5,
 		};
 
 		const nodeVersions: IVersionedNodeType['nodeVersions'] = {
 			1: new HttpRequestV1(baseDescription),
 			2: new HttpRequestV2(baseDescription),
 			3: new HttpRequestV3(baseDescription),
 			4: new HttpRequestV3(baseDescription),
+			5: new HttpRequestV3(baseDescription),
```
Comment (netroy) [inline packages/nodes-base/nodes/HttpRequest/HttpRequest.node.ts:25]: considering that v4 was added only 3 days ago, and isn't out in any release, maybe we should use v4 for this check as well.

**Comment by csuermann** [comment]
> With this PR if the toggle is not in the UI it will follow redirects, however if you add the toggle it will already be set as false.

My point is: As the toggle already exists, shouldn't it be also visibly switched on in the UI to clearly express the new default behavior? It probably shouldn't be an optional property either then, no?

**Review by netroy — DISMISSED** [review]
If I add the `Redirects` option, the flag is correctly enabled, and redirects work as expected. But, when the option isn't added to the node, the behavior is defaulting to the same as in v1-3.

**Review by netroy — DISMISSED** [review]
LGTM

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@0.223.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%400.223.0)

**Pull request merged**